declaring type alias of template type causes c error: ';' expected (got "main__Foo")

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.8 18eee34.04aba8f
OS: windows, Microsoft Windows 11 Pro v22631 64-bit
Processor: 24 cpus, 64bit, little endian, 

getwd: D:\vbug2
vexe: C:\v\v.exe
vexe mtime: 2024-09-30 05:47:23

vroot: OK, value: C:\v
VMODULES: OK, value: C:\Users\info\.vmodules
VTMP: OK, value: C:\Users\info\AppData\Local\Temp\v_0

Git version: git version 2.42.0.windows.2
Git vroot status: 0.4.8-7-g04aba8f7-dirty
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,

operable program or batch file.


thirdparty/tcc status: thirdparty-windows-amd64 b425ac82

```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg main.v`
```v

module main

type Foo = T

fn main() {}
```

**What did you expect to see?**

no c compiler error


**What did you see instead?**
```
================== C compilation error (from tcc): ==============
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01J9168JQG0DBRB0VD2M1M5P0K.tmp.c:526: warning: WINVER redefined
cc: C:/Users/info/AppData/Local/Temp/v_0/main.01J9168JQG0DBRB0VD2M1M5P0K.tmp.c:1022: error: ';' expected (got "main__Foo")
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -533,7 +533,7 @@ fn (mut c Checker) alias_type_decl(node ast.AliasTypeDecl) {
 		c.error('cannot make an alias of Result type', node.type_pos)
 	}
 	match parent_typ_sym.kind {
-		.placeholder, .int_literal, .float_literal {
+		.placeholder, .int_literal, .float_literal, .any {
 			c.error('unknown aliased type `${parent_typ_sym.name}`', node.type_pos)
 		}
 		.alias {

--- a/vlib/v/checker/tests/alias_to_generic_err.out
+++ b/vlib/v/checker/tests/alias_to_generic_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/alias_to_generic_err.vv:4:12: error: unknown aliased type `T`
+    2 | module main
+    3 | 
+    4 | type Foo = T
+      |            ^
+    5 | 
+    6 | fn main() {

--- a/vlib/v/checker/tests/alias_to_generic_err.out
+++ b/vlib/v/checker/tests/alias_to_generic_err.out
@@ -1,7 +1,7 @@
-vlib/v/checker/tests/alias_to_generic_err.vv:4:12: error: unknown aliased type `T`
-    2 | module main
-    3 | 
-    4 | type Foo = T
+vlib/v/checker/tests/alias_to_generic_err.vv:3:12: error: unknown aliased type `T`
+    1 | module main
+    2 | 
+    3 | type Foo = T
       |            ^
-    5 | 
-    6 | fn main() {
+    4 | 
+    5 | fn main() {

--- a/vlib/v/checker/tests/alias_to_generic_err.vv
+++ b/vlib/v/checker/tests/alias_to_generic_err.vv
@@ -1,0 +1,6 @@
+module main
+
+type Foo = T
+
+fn main() {
+}

--- a/vlib/v/checker/tests/any_type_err.out
+++ b/vlib/v/checker/tests/any_type_err.out
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/any_type_err.vv:3:16: error: cannot use type `any` here
+vlib/v/checker/tests/any_type_err.vv:3:16: error: unknown aliased type `any`
     1 | // Any types should error, while parametrically polymorphic should not.
     2 | 
     3 | type AnyType = any
@@ -19,6 +19,13 @@ vlib/v/checker/tests/any_type_err.vv:5:27: error: cannot use type `any` here
       |                           ~~~
     6 | 
     7 | type PolyType = T
+vlib/v/checker/tests/any_type_err.vv:7:17: error: unknown aliased type `T`
+    5 | type AnyPolySumType = T | any
+    6 | 
+    7 | type PolyType = T
+      |                 ^
+    8 | type PolySumType = T | string
+    9 |
 vlib/v/checker/tests/any_type_err.vv:11:6: error: cannot use type `any` here
     9 | 
    10 | struct AnyStructField[T] {

